### DISCUSSION TITLE
Unable to correctly limit subscription video download by time or by max downloads

### DISCUSSION THREAD

**Discussion by sn4201** [jmbannon/ytdl-sub#813] [Q&A]
Struggling here, i'm using a subscription example file that uses "only recent" preset, however if I go to the config file and try to use max_downloads to limit the number of downloads, it doesn't respect the limit.  I'm not sure what preset settings "only recent" is using, if i look in the config file i'm not sure I understand what "recent" is referring to.  I would like to cap the total number of downloads and have that take precedence over the date range if possible.  However it seems like when  I try to make a new preset to have a different date range for different channels (some channels upload daily others upload weekly, would prefer not to download a whole month of a daily channel), I get various syntax errors.  

I think this would be a great app if it were a little more user friendly and easy to use, but i suppose things worth doing are rarely easy...anyway, i appreciate the effort in the program. But I hope in time it will get further development in terms of usability, because unfortunately i have spent several hours dry running various config and subscription yaml files and can't seem to get anything to work the way i want.  Probably user error involved but i just dont have the time to keep dedicating to this unfortunately.  Will be following the project, i hope things continue to improve!

**Comment by jmbannon** [comment]
Can you upload your subscription file

**Reply by jmbannon** [reply]
When I get home I can add to the examples on how to do this - it should be doable without any config, but does require knowing the syntax in-depth which to your point is not very new-user friendly

**Reply by sn4201** [reply]
```# subscriptions.yaml

# global overrides for all subscriptions
__preset__:
  overrides:
    tv_show_directory: "/tv_shows"
    date_range: "1months"

# All subs under this use the `Plex TV Show by Date` preset
Kodi TV Show by Date:

  # Uses `Only Recent` preset to keep 2months worth
  Only Recent:
    "Dustups": "https://www.youtube.com/@Dustupstexas"
    "Life For Sale":  "https://www.youtube.com/@BenMallah"

#  Only 1wk:
#    "CBC News The National":  "https://www.youtube.com/playlist?list=PLvntPLkd9IMcbAHH-x19G85v_RE-ScYjk"
#    "Aaron Bennett": "https://www.youtube.com/@abennett/"
```

I tried to create a 1wk preset in the config file called 1wk_videos and it was breaking no matter how i tried to call it from the subscriptions file. also tried editing the presets in the config file to add max downloads in various places and that also didnt work.  seems like right now the subscriptions/config file im using downloads the last month or so but in some cases thats way too much

Heres the config, copied from the walkthrough mostly IIRC:

``` # config.yaml

configuration:
  working_directory: '/tv_shows/dl'
  dl_aliases:
    video: "--preset video"
    video-url: "--download.url"
    tv-show-name: "--overrides.tv_show_name"
    single-music-video: "--preset video --overrides.tv_show_name 'Music Video' --download.url"

presets:
  video:
    output_options:
      output_directory: "/tv_shows/{tv_show_name_sanitized}"
      file_name: "{episode_path}.{ext}"
      thumbnail_name: "{episode_path}-thumb.{thumbnail_ext}"

    overrides:
      tv_show_name: "OVERRIDE IN SUBSCRIPTION"
      season: "{upload_year}"
      episode: "{upload_month}{upload_day_padded}"
      episode_padded: "{upload_month_padded}{upload_day_padded}"
      episode_file_name: "s{season}e{episode_padded}.{title_sanitized}"
      episode_path: "Season {season}/{episode_file_name}"

    nfo_tags:
      nfo_name: "{episode_path}.nfo"
      nfo_root: "episodedetails"
      tags:
        title: "{title}"
        season: "{season}"
        episode: "{episode}"
        year: "{upload_year}"
        aired: "{upload_date_standardized}"

    output_directory_nfo_tags:
      nfo_name: "tvshow.nfo"
      nfo_root: "tvshow"
      tags:
        title: "{tv_show_name}"

  playlist:
    preset: "video"
    output_options:
      maintain_download_archive: True
      keep_files_after: now-2weeks
    ytdl_options:
      break_on_existing: True
      max_downloads: 6


  channel:
    preset: "playlist"
    download:
       playlist_thumbnails:
        - name: "poster.jpg"
          uid: "avatar_uncropped"
        - name: "banner.jpg"
          uid: "banner_uncropped"

  recent_videos:
    date_range:
      after: "now-1months"
    output_options:
      keep_files_after: "now-1months"
    ytdl_options:
      max_downloads: 6

  1wk_videos:
    date_range:
      after: "now-1weeks"
    output_options:
      keep_files_after: "now-1weeks"
```

**Reply by jmbannon** [reply]
You are very close. I would delete everything in the config except 1wk_videos. Then rename 1wk_videos to "Only 1wk". That should work with your current subscriptions

**Comment by jmbannon** [comment]
If you want to set date_range for a single subscription, you can also do this:
```
Only Recent:
    "~Dustups":
        url: "https://www.youtube.com/@Dustupstexas"
        date_range: "3weeks"
```
The ~ means 'set override variables underneath it' and will override anything in `__preset__` and other presets

**Reply by sn4201** [reply]
This is working well, thank you.  This way i can have the default download for about 1mo, and any exceptions i can set per channel.

So the way I have things set now, will the system automatically remove old files to keep within the date range?

Also, I'm noticing using my current configs, Kodi is displaying episodes in the show list as: <year>x<mo><day> - <year>-<mo><day> - <Title> so there seems to be a duplication happening somewhere.  Can you suggest any adjustments to avoid this?  Is this because I copied the config from the walkthrough maybe?

Thank you for your assistance, i appreciate the help

**Reply by jmbannon** [reply]
> So the way I have things set now, will the system automatically remove old files to keep within the date range?

Correct.

> Kodi is displaying episodes in the show list as: x - - - <Title> 

What does `x - - - <Title> ` mean - can you take a screenshot? I thought for duplicates, Kodi would add a `(1)` or `(2)` suffix.
If you downloaded anything from the walkthrough, I would delete it - it was only to service as a learning guide.

If so and worst case, just wipe all the files for the channel/playlist you have issue with and redownload using the prebuilt presets (assuming you did use the walkthrough guide presets)

**Reply by sn4201** [reply]
Sorry, some of the fields in brackets got removed from the post.

Basically its writing the list as: (Year)x(mo)(day padded) - (year)(mo)(day) - (Title)

**Reply by jmbannon** [reply]
The first is the episode number, then the second is part of the title (the date is included by default). You can change this by setting the episode_title variable

**Reply by sn4201** [reply]
Where/how do I set that?  I dont see episode_title variable in the existing config files I posted above.

**Reply by jmbannon** [reply]
Just like date_range: https://github.com/jmbannon/ytdl-sub/blob/master/src/ytdl_sub/prebuilt_presets/tv_show/episode.yaml#L40

**Reply by sn4201** [reply]
Thank you, was able to set it to just (title) and for everything and now it doesn't duplicate the date.  I appreciate your assistance
